ci-operator/step-registry/cucushift/installer/rehearse/azure/cases/multi-clusters/same-subnets/provision/cucushift-installer-rehearse-azure-cases-multi-clusters-same-subnets-provision-commands.sh

```diff
@@ -0,0 +1,374 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# save the exit code for junit xml file generated in step gather-must-gather
+# pre configuration steps before running installation, exit code 100 if failed,
+# save to install-pre-config-status.txt
+# post check steps after cluster installation, exit code 101 if failed,
+# save to install-post-check-status.txt
+EXIT_CODE=101
+trap 'save_artifacts; if [[ "$?" == 0 ]]; then EXIT_CODE=0; fi; echo "${EXIT_CODE}" > "${SHARED_DIR}/install-post-check-status.txt"' EXIT TERM
+
+function save_artifacts()
+{
+    set +o errexit
+    cp "${install_dir}/metadata.json" "${SHARED_DIR}/cluster-2-metadata.json"
+    cp "${install_dir}"/log-bundle-*.tar.gz "${ARTIFACT_DIR}/" 2>/dev/null
+
+    current_time=$(date +%s)
+    sed '
+      s/password: .*/password: REDACTED/;
+      s/X-Auth-Token.*/X-Auth-Token REDACTED/;
+      s/UserData:.*,/UserData: REDACTED,/;
+      ' "${install_dir}/.openshift_install.log" > "${ARTIFACT_DIR}/cluster_2_openshift_install-${current_time}.log"
+
+    if [ -d "${install_dir}/.clusterapi_output" ]; then
+        mkdir -p "${ARTIFACT_DIR}/cluster_2_clusterapi_output-${current_time}"
+        cp -rpv "${install_dir}/.clusterapi_output/"{,**/}*.{log,yaml} "${ARTIFACT_DIR}/cluster_2_clusterapi_output-${current_time}" 2>/dev/null
+    fi
+
+    set -o errexit
+}
+
+function scale_up_worker_nodes()
+{
+    local machineset_name replicas total_nodes_count ret=0
+    total_nodes_count=$(oc get node --no-headers | grep worker | wc -l)
+    machineset_name=$(oc -n openshift-machine-api get -o "jsonpath={range .items[*]}{.metadata.name}{'\\n'}{end}" machinesets | grep worker | head -n1)
+    replicas=$(oc get machineset ${machineset_name} -n openshift-machine-api -o "jsonpath={.spec.replicas}")
+
+    echo "scale up worker on machineset ${machineset_name}..."
+    timeout 5m oc scale --replicas=$((replicas+1)) machineset ${machineset_name} -n openshift-machine-api || ret=1
```

Comment:
I guess we can merge target replicas and total_nodes_count as one variable.

Reply:
there will be several machinesets created if installing cluster in region with zone supported ( 1 machineset per zone). In this case, replicas does not equal to total_nodes_count.

Reply:
Ah, you are right, I forgot it.